standalone-dns-proxy/cmd/standalonednsproxy.go

```diff
@@ -16,24 +24,109 @@ type StandaloneDNSProxy struct {
 	standaloneDNSProxyServerPort uint16
 	enableL7Proxy                bool
 	enableStandaloneDNSProxy     bool
+	dnsProxier                   proxy.DNSProxier
+
+	// connHandler is the gRPC connection handler client for standalone DNS proxy
+	connHandler client.ConnectionHandler
+
+	// dnsRulesTable is the table that holds DNS rules received from the Cilium agent
+	// It is used by the DNS proxy to enforce DNS policies
+	dnsRulesTable statedb.RWTable[client.DNSRules]
```

Comment:
I wonder if we should look into just doing a generic gRPC (or just Swagger REST) service for synchronizing a StateDB table from one process to another.
It would essentially just be a stream of `statedb.Change`s and on the client side we'd just do bit of batching + WriteTxn + N Insert + Commit. And if we lose the stream we'd just start over from scratch (e.g. `DeleteAll` and do `Changes()` again from start).

This would save fair bit of boilerplate in SDP as we wouldn't need all the RPC definitions and handlers and their tests and so on.